config/routes.rb

```diff
@@ -5,15 +5,26 @@
   # You can have the root of your site routed with "root"
   root 'task#index'
 
-  get "/task_:id/details" => "task#show"
+  post "/task_:id/mark_complete"   => "task#mark_complete"
+  post "/task_:id/mark_incomplete" =>      "task#mark_incomplete"
 
-  get "/task/new" => "task#new"
-  post "/task/new" => "task#create"
+  get "/task_:id/details"         => "task#show"
```

Comment:
It's a common REST pattern to include the type of the resource in the route. This helps distinguish between showing a task and showing a person (which you aren't specifically doing, but you might do someday). The pattern that Rails uses is uses the plural of the resource in the path, so something like this:

```ruby
get "/tasks/"         => "task#index"
get "/tasks/new"      => "task#new"
post "/tasks/"        => "task#create"
get "/tasks/:id"      => "task#show"
get "/tasks/:id/edit" => "task#edit"
patch "/tasks/:id"    => "task#update"
delete "/tasks/:id"   => "task#destroy"
```